source/connect/csot.txt

```diff
@@ -1,210 +1,218 @@
-       await coll.insert_one({"name": "Nunu"})
-       await coll.find_one({"name": "Nunu"})
+      .. code-block:: python
 
-Troubleshooting
----------------
+         uri = "mongodb://<username>:<password>@<hostname@:<port>/?timeoutMS=10000"
+         client = pymongo.MongoClient(uri)
 
-In code, you can identify timeout errors by using the ``pymongo.errors.PyMongoError.timeout``
-property. Some specific timeout errors examples are described below.
+         coll = client.test.test
+         coll.insert_one({"name": "Yngwie"})  # Uses a 10-second timeout.
+         coll.find_one({"name": "Yngwie"})  # Also uses a 10-second timeout.
 
-When the client was unable to find an available server to run the operation
-within the given timeout:
+   .. tab:: timeout Method
+      :tabid: method
 
-.. code-block:: python
+      .. code-block:: python
 
-   pymongo.errors.ServerSelectionTimeoutError: No servers found yet, Timeout: -0.00202266700216569s, Topology Description: <TopologyDescription id: 63698e87cebfd22ab1bd2ae0, topology_type: Unknown, servers: [<ServerDescription ('localhost', 27017) server_type: Unknown, rtt: None>]>
+         client = MongoClient()
+         coll = client.test.test
+         with pymongo.timeout(10):
+            coll.insert_one({"name": "Yngwie"})
+         with pymongo.timeout(10):
+            coll.find_one({"name": "Yngwie"})
 
-When either the client was unable to establish a connection within the given
-timeout or the operation was sent but the server was not able to respond in time:
+.. important:: ``timeout()`` Overrides ``timeoutMS``
 
-.. code-block:: python
+   If you specify the ``timeoutMS`` option, then call the ``timeout()`` method
+   in your code, {+driver-short+} ignores the ``timeoutMS`` value inside the ``timeout``
+   block:
 
-   pymongo.errors.NetworkTimeout: localhost:27017: timed out
+   .. code-block:: python
 
-When the server cancelled the operation because it exceeded the given timeout.
-Note that the operation may have partially completed on the server (depending
-on the operation).
+      client = MongoClient("mongodb://localhost/?timeoutMS=10000")
+      coll = client.test.test
+      coll.insert_one({"name": "Yngwie"})  # Uses the client's 10-second timeout
+      # pymongo.timeout overrides the client's timeoutMS.
+      with pymongo.timeout(20):
+         coll.insert_one({"name": "Yngwie"})  # Uses the 20-second timeout
+      with pymongo.timeout(5):
+         coll.find_one({"name": "Yngwie"})  # Uses the 5-second timeout
 
-.. code-block:: python
+Handling Exceptions
+-------------------
 
-   pymongo.errors.ExecutionTimeout: operation exceeded time limit, full error: {'ok': 0.0, 'errmsg': 'operation exceeded time limit', 'code': 50, 'codeName': 'MaxTimeMSExpired'}
+When a server operation exceeds the specified timeout, {+driver-short+}
+raises a timeout exception. If the exception is the result of a timeout, the
+``PyMongoError.timeout`` property is ``True``. Otherwise, it's ``False``.
```

Comment:
This is a bit confusing. If the driver raises a timeout exception, won't the exception always be the result of a timeout, and therefore always have the property as `True`?

I think this could be more clear if we just focused on handling the timeout error specifically and don't talk in the narrative about when it's false (can keep that else block in the code example though, just maybe un-highlight it). 

Happy to discuss this more if I'm misunderstanding or anything.

```suggestion
When a server operation exceeds the specified timeout, {+driver-short+}
raises a timeout exception, and sets the
``PyMongoError.timeout`` exception property to ``True``. 
```

Reply:
yeah, i can see the confusion. reworded